bfsdk/include/bftypes.h

```diff
@@ -135,4 +135,22 @@ typedef INT_PTR intptr_t;
 #define PRId64 "lld"
 #endif
 
+/* -------------------------------------------------------------------------- */
+/* VMM Types                                                                  */
+/* -------------------------------------------------------------------------- */
+
+#ifdef __cplusplus
+#include <bfgsl.h>
+#include <bfarch.h>
+#if defined(BF_INTEL_X64)
+namespace bfvmm::intel_x64
+{
+class domain;
+class vcpu;
+}
+using domain_t = gsl::not_null<bfvmm::intel_x64::domain *>;
```

Comment:
I removed these types from the vcpu and domain classes to get around some circular includes that occur with the new delegator pattern. I'd be ok to move them somewhere else, but I don't think we can get around the circular include if we move them back to the vcpu and domain classes.